sdmetrics/multi_table/multi_single_table.py

```diff
@@ -67,22 +67,28 @@ def _compute(self, real_data, synthetic_data, metadata=None):
         elif not isinstance(metadata, dict):
             metadata = metadata.to_dict()
 
-        values = []
+        scores = {}
         errors = {}
         for table_name, real_table in real_data.items():
             synthetic_table = synthetic_data[table_name]
             table_meta = metadata['tables'][table_name]
 
             try:
-                score = self.single_table_metric.compute(real_table, synthetic_table, table_meta)
-                values.append(score)
-            except IncomputableMetricError as error:
+                if hasattr(self.single_table_metric, 'compute_breakdown'):
+                    score_breakdown = self.single_table_metric.compute_breakdown(
+                        real_table, synthetic_table, table_meta)
+                    scores[table_name] = score_breakdown
+                else:
+                    score = self.single_table_metric.compute(
+                        real_table, synthetic_table, table_meta)
+                    scores[table_name] = score
```

Comment:
instead of this check could we catch the attribute error and do compute instead and then have the other exception block you have to catch everything else?